web-app/src/pages/Dashboard/container.js

```diff
@@ -219,25 +426,23 @@ const DashboardContainer = Wrapped =>
         this.getTakenHolidays(userDetails.employeeId);
         this.closeModal();
       });
-    }
+    };
 
     render() {
       return (
         this.state.userDetails &&
         this.state.takenHolidays && (
           <Wrapped
+            formElementsArray={getFormElementsArray(this.state.booking.form)}
             onSelectSlot={this.onSelectSlot}
             onSelectEvent={this.onSelectEvent}
             closeModal={this.closeModal}
-            changeStart={this.changeStart}
-            changeEnd={this.changeEnd}
-            changeHalfday={this.changeHalfday}
+            formChanged={(event, id) => this.handleFormChange(event, id)}
             userDetails={this.state.userDetails}
             requestHoliday={this.submitHolidayRequest}
             takenHolidays={this.state.takenHolidays}
             updateHoliday={() => this.updateHoliday(false)}
             cancelHoliday={() => this.updateHoliday(true)}
-            changeWFH={this.changeWFH}
             {...this.state}
             {...this.props}
           />
```

Comment:
This isn't really a concern for this PR, but holy crap! A ~450 line JS file!

I think this is an indication that we need to review how we're composing components.

Reply:
Totally agree. I'm new to react so any guidance would be greatly appreciated. ps this was a large file before I came along.